### PR TITLE
feat: Resolve Instance Types into the NodePool Status

### LINKED ISSUE #611: Resolve Instance Types into the NodePool Status
### Description

**What problem are you trying to solve?**

Right now, we resolve InstanceTypes on a per-NodePool basis. This is done by making calls out to the `GetInstanceTypes()` API and then layering NodePool requirements on top of these retrieved InstanceTypes to store details about which InstanceTypes are supported for a given NodePool. This is further layered to consider the pod workload requirements and offering availability during scheduling.

We'd like to be able to store all available instance types that were resolved from the NodePool requirements into the NodePool status for observability. This would help users who don't know how diverse or constrained they are being from instance types get a better undrestanding.

It might look something like this

```yaml
apiVersion: karpenter.sh/v1beta1
kind: NodePool
spec:
...
status:
  instanceTypes:
  - instance-type-a
  - instance-type-b
  - instance-type-c
```

We _probably_ don't want to store offering details into the status since that might get too noisy and difficult to reason about.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by sadath-12**
Fixes #611  

**Description**

InstanceTypes that were resolved considering the compatiblity , fits , offerings can now be viewed on nodepool status 
as described in the issue 

for example: 

```
apiVersion: karpenter.sh/v1beta1
kind: NodePool
spec:
...
status:
  instanceTypes:
  - instance-type-a
  - instance-type-b
  - instance-type-c
  ```

**How was this change tested?**

```make presubmit```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by njtran** [comment]
Hey @sadath-12, I marked this PR as needing a design as this is API surface that we should review. Would you be able to create a design doc exploring some of the details that you explored here? It looks like you've added the string names of the instance types, but haven't included any requirements or availability.

It'd probably be best figured out through a design doc to see which parts of these instance types we want to include, as we don't want the NodePool Status to be too large that it's hard to parse useful information from the status. Even at the moment, for a default NodePool, this would return almost all instance types from the `aws ec2 describe-instance-types` call, which would be hundreds of lines, which may not be what the user wants. Have you thought about how we should structure this?

**Comment by sadath-12** [comment]
Not yet @njtran but I'm open to what's to be done
